core/app/[locale]/store-selector/page.tsx

```diff
@@ -0,0 +1,76 @@
+import { getTranslations, unstable_setRequestLocale } from 'next-intl/server';
+
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { Link } from '~/components/link';
+import { StoreLogo, StoreLogoFragment } from '~/components/store-logo';
+import { NavigationMenu, NavigationMenuLink } from '~/components/ui/navigation-menu';
+import { locales, LocaleType } from '~/i18n';
+
+import { LocaleLink } from './_components/locale-link';
+
+export const metadata = {
+  title: 'Location selector',
+};
+
+const StoreSelectorPageQuery = graphql(
+  `
+    query LocationSelectorPageQuery {
+      site {
+        settings {
+          ...StoreLogoFragment
+        }
+      }
+    }
+  `,
+  [StoreLogoFragment],
+);
+
+export default async function StoreSelector({
+  params: { locale: selectedLocale },
+}: {
+  params: { locale: LocaleType };
+}) {
+  unstable_setRequestLocale(selectedLocale);
+
+  const t = await getTranslations('StoreSelector');
+
+  const { data } = await client.fetch({
+    document: StoreSelectorPageQuery,
+  });
+
+  const storeSettings = data.site.settings;
+
+  return (
+    <>
+      <header>
+        <NavigationMenu>
+          <NavigationMenuLink asChild>
+            {storeSettings && (
+              <Link className="p-0" href="/">
+                <StoreLogo data={storeSettings} />
+              </Link>
+            )}
+          </NavigationMenuLink>
+        </NavigationMenu>
+      </header>
+
+      <div className="flex flex-col gap-2 px-4 lg:container sm:px-10 lg:mx-auto lg:max-w-[1000px] lg:px-12">
+        <h1 className="text-3xl font-black lg:text-4xl">{t('heading')}</h1>
+
+        <div className="grid grid-cols-1 gap-6 py-6 md:grid-cols-3 md:gap-11 lg:grid-cols-4 lg:gap-8">
+          {locales.map((locale) => (
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+            <LocaleLink key={locale} locale={locale} selected={selectedLocale === locale} />
```

Comment:
Similar issue here, unfortunately had to prevent this rule from applying because when it is only one locale typescript doesn't see the need for this conditional since selectedLocale can only match the current locale.